test/unit/query/builder.js

```diff
@@ -9863,4 +9863,31 @@ describe('QueryBuilder', () => {
       }
     );
   });
+
+  it('#3553 should correctly compile insert with BigInt', function() {
+    testquery(
+      qb()
+        .insert([{ a: BigInt(12) }])
```

Comment:
`BigInt("1234567182348761238746512348765")` just to make sure it's not coercing the values back into regular `Number` instances.